lib/logger.js

```diff
@@ -153,32 +156,50 @@ Logger.prototype._bufferLog = function(message) {
 
     this._bufByteLength += sizeof(message);
 
+    debug('Buffering message: %s', message.line);
     this._buf.push(message);
 
     if (this._bufByteLength >= this._flushLimit) {
+        debug('Buffer size meets (or exceeds) flush limit.  Immediately flushing');
         this._flush();
         return;
     }
 
     if (!this._flusher) {
+        debug('No scheduled flush. Scheduling for %d ms from now.', configs.FLUSH_INTERVAL);
         this._flusher = setTimeout(bind(this._flush, this), configs.FLUSH_INTERVAL);
     }
 };
 
 Logger.prototype._flush = function(cb) {
-    if (!this._req || this._buf.length === 0) { cb && cb(); return; }
+    if (!this._req || this._buf.length === 0) {
+        debug('Nothing to flush');
+        cb && cb();
+        return;
+    }
 
     var sendbuf = { e: 'ls', ls: this._buf };
     var data = stringify(sendbuf);
 
+    // BEFORE we clear the buffer, capture the lines being flushed for debug output
+    var dbgLines = this._buf.map(function(msg) { return msg.line; });
```

Comment:
Have to do this early because `this._buf` is emptied before making the request.  Also it may have changed by the time the api responds and we want to capture the value it has *now*.